src/layer/marker/DivIcon.js

```diff
@@ -33,6 +33,10 @@ export var DivIcon = Icon.extend({
 		// Custom HTML code to put inside the div element, empty by default.
 		html: false,
 
+		// @option element: DOM element = ''
```

Comment:
The idea is good, but I think that the approach is wrong. Adding one more option might be confusing.

Instead, look at the API surface of `L.Popup`: there is a `setContent()` method which accepts **either** a string (of HTML text), or an instance of `HTMLElement`. The code then checks the type of that parameter and [either sets `innerHTML` or appends the child element](https://github.com/Leaflet/Leaflet/blob/master/src/layer/DivOverlay.js#L167).

I think the approach we currently follow in `L.Popup` (and `L.Tooltip`) is cleaner than having two parallel ways of adding content. What would you think about following this pattern instead?